cabal-install/src/Distribution/Client/Init/NonInteractive/Command.hs

```diff
@@ -274,14 +274,16 @@ licenseHeuristics :: Interactive m => InitFlags -> m SpecLicense
 licenseHeuristics flags = getLicense flags $ guessLicense flags
 
 -- | The author's name. Prompt, or try to guess from an existing
---   darcs repo.
+--   git repo.
 authorHeuristics :: Interactive m => InitFlags -> m String
-authorHeuristics flags = getAuthor flags guessAuthorEmail
+authorHeuristics flags = guessAuthorEmail >>=
+  maybe (getAuthor flags $ return "") (getAuthor flags . return)
```

Comment:
Thinking more about, maybe having an empty string in the resulting file would be confusing. Can we put something generic there? Like "Unknown" or "Anonymous". Same for email.

Reply:
"Unknown" sounds best to me